pairing/bls12381/point.go

```diff
@@ -0,0 +1,403 @@
+package bls12381
+
+import (
+	"crypto/cipher"
+	"crypto/sha256"
+	"encoding/hex"
+	"io"
+
+	bls12381 "github.com/kilic/bls12-381"
+	"go.dedis.ch/kyber/v3"
+	"go.dedis.ch/kyber/v3/group/mod"
+)
+
+// pointG1 is a kyber.Point holding a G1 point on BLS12-381 curve
+type pointG1 struct {
+	p *bls12381.PointG1
+}
+
+func nullpointG1() *pointG1 {
+	var p bls12381.PointG1
+	return newPointG1(&p)
+}
+
+func newPointG1(p *bls12381.PointG1) *pointG1 {
+	return &pointG1{p: p}
+}
+
+func (k *pointG1) Equal(k2 kyber.Point) bool {
+	return bls12381.NewG1().Equal(k.p, k2.(*pointG1).p)
+}
+
+func (k *pointG1) Null() kyber.Point {
+	return newPointG1(bls12381.NewG1().Zero())
+}
+
+func (k *pointG1) Base() kyber.Point {
+	return newPointG1(bls12381.NewG1().One())
+}
+
+func (k *pointG1) Pick(rand cipher.Stream) kyber.Point {
+	var dst, src [32]byte
+	rand.XORKeyStream(dst[:], src[:])
+	return k.Hash(dst[:])
+}
+
+func (k *pointG1) Set(q kyber.Point) kyber.Point {
+	k.p.Set(q.(*pointG1).p)
+	return k
+}
+
+func (k *pointG1) Clone() kyber.Point {
+	var p bls12381.PointG1
+	p.Set(k.p)
+	return newPointG1(&p)
+}
+
+func (k *pointG1) EmbedLen() int {
+	panic("bls12-381: unsupported operation")
+}
+
+func (k *pointG1) Embed(data []byte, rand cipher.Stream) kyber.Point {
+	panic("bls12-381: unsupported operation")
+}
+
+func (k *pointG1) Data() ([]byte, error) {
+	panic("bls12-381: unsupported operation")
+}
+
+func (k *pointG1) Add(a, b kyber.Point) kyber.Point {
+	aa := a.(*pointG1)
+	bb := b.(*pointG1)
+	bls12381.NewG1().Add(k.p, aa.p, bb.p)
+	return k
+}
+
+func (k *pointG1) Sub(a, b kyber.Point) kyber.Point {
```

Comment:
Should be changed accordingly to @nkcr's suggested change for `Add`. Same for the remaining functions.